pallets/proof_of_existence/src/benchmarking.rs

```diff
@@ -13,23 +13,71 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#![cfg(feature = "runtime-benchmarks")]
+
 use super::*;
 
-use frame_benchmarking::{benchmarks, impl_benchmark_test_suite};
+use frame_benchmarking::v2::*;
 use frame_system::RawOrigin;
 use sp_core::{Get, H256};
+use sp_runtime::SaturatedConversion;
+
+#[benchmarks]
+mod benchmarks {
+    use super::*;
+
+    const OFFSET_FACTOR: u64 = 10000;
+    #[benchmark]
+    fn publish_attestation() {
+        // Setup code
+        // benchmark taking into account situations where attestations left to be cleaned in storage
+        // are bigger than one. shouldn't happen very often so we consider only 0.01% of the
+        // maximum supported ones.
+        let max_attestations = T::MaxStorageAttestations::get().saturated_into::<u64>();
+        let offset = max_attestations
```

Comment:
I think you'r overestimate here: in this case you're considering the case where you're pruning (10 + 1) full attested tree with 32 leafs for each case. As far as I can understand you can have just one attestation to prune (32 proofs in the wroste case)